mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -2354,9 +2427,11 @@ def list_scorer_versions(self, experiment_id, name) -> list[ScorerVersion]:
                 )
 
             # Convert to mlflow.entities.scorer.ScorerVersion objects
-            return [
+            entities = [
```

Comment:
nit: we probably should convert to `ScorerVersion` **after** overwriting the model property since the `serialized_scorer` method of `ScorerVersion` is cached and we might mistakenly call this method that breaks this gateway uri logic.